### PR TITLE
Add ability to change log_level at runtime to system-probe

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

This PR adds a `config` subcommand, like the core agent, which allows changing specific configuration values at runtime. This PR adds the ability to change the `log_level` config value for system-probe.

### Motivation

Increased need to debug system-probe on production systems without restarting.

### Additional Notes

- The runtime setting code was moved from `cmd/agent/app/settings` to `pkg/config/settings` so that other agents could re-use it.
- The system-probe is communicated to over the existing unix socket on Linux, rather than TCP like the core agent.

### Describe your test plan

1. Start system-probe
2. Get current `log_level` with `system-probe config get log_level`
3. Change `log_level` with `system-probe config set log_level debug` (or another log level value)
4. Notice the change in level via the system-probe logs.

### CODE REVIEW

File: cmd/system-probe/app/config.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	if err != nil {
+		return err
+	}
+
+	hidden, err := c.Set(args[0], args[1])
+	if err != nil {
+		return err
+	}
+	if hidden {
+		fmt.Printf("IMPORTANT: you have modified a hidden option, this may incur in billing or other unexpected side-effects.\n")
```
Comment (heyronhay) [inline cmd/system-probe/app/config.go:130]: "incur in billing" -> "incur charges in billing"?  Seems like a word was left out :)

**Review by heyronhay — APPROVED** [review]

File: cmd/agent/app/config.go [hunk 2]
```diff
@@ -75,162 +71,106 @@ func setupConfig() error {
 	return util.SetAuthToken()
 }
 
-func showRuntimeConfiguration(cmd *cobra.Command, args []string) error {
+func getClient() (commonsettings.Client, error) {
```
Comment (truthbk) [inline cmd/agent/app/config.go:74]: nit: `getSettingsClient()` maybe?

File: cmd/system-probe/app/config.go (hunk 1 above)
Comment (brycekahle) [reply cmd/system-probe/app/config.go]: Yeuuup. Fixed.

File: cmd/agent/app/settings/runtime_settings_test.go [hunk 3]
```diff
@@ -1,113 +1,17 @@
-// Unless explicitly stated otherwise all files in this repository are licensed
```
Comment (truthbk) [inline cmd/agent/app/settings/runtime_settings_test.go:1]: Why remove the copyright header? This applies to a few other files, I think we want this everywhere.

**Review by truthbk — APPROVED** [review]
Looks great! Just very minor nits (mostly just about naming) - I think you can ignore most of the concerns if you're not happy making the changes.

File: cmd/agent/app/settings/runtime_settings_test.go (hunk 3 above)
Comment (brycekahle) [reply cmd/agent/app/settings/runtime_settings_test.go:1]: This file was moved and then part of it moved back, which is how it lost the header. I'll restore it.

**Review by truthbk — APPROVED** [review]
Unsure why the tests are failing, but this looks ready for 🚀  in my book.

**Review by hmahmood — COMMENTED** [review]
This looks fine to me overall, but I'm not sure why the rename from `api` -> `module` is part of this PR. Maybe that should be a follow on?

**Comment by sgnn7** [comment]
@brycekahle This PR should target 7.29 milestone or get into the codebase asap as we have entered the freeze period for the release.

File: cmd/system-probe/modules/all_unsupported.go [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 package modules
 
-import "github.com/DataDog/datadog-agent/cmd/system-probe/api"
+import "github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
```
[the remark this answers is not in the document]
Comment (brycekahle) [reply cmd/system-probe/modules/all_unsupported.go:5]: Yes. Otherwise there is an import loop.

**Review by hmahmood — APPROVED** [review]

**Review by Gui774ume — APPROVED** [review]
lgtm for the Runtime Security related files 👍🏻

**Pull request merged**